connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -220,6 +218,11 @@ private String startConnector(Map<String, String> connectorProps) {
             state.configOriginals = connectorProps;
             state.config = connConfig;
         }
+
+        if (tempConnectors.containsKey(connName)) {
```

Comment:
Same question here as with the DistributedHerder -- what is this accomplishing and why is it needed before creating the connector? I'm not even sure this is effective since requests for validation can come in in parallel with a connector starting.